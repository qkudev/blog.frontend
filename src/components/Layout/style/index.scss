@import '../../../style/variables';

.layout {
  min-height: 100vh;
  min-width: 100vw;
  padding: 0 1rem;

  @media (min-width: 768px) {
    padding: 0 3rem;
  }

  &.content {
    position: absolute;
    padding-top: 3rem;
    background-color: $body-bg-color;
    z-index: 1000;
  }
}
