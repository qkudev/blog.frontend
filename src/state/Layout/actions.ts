type LoadingBegin = IAction<'LAYOUT_LOADING_BEGIN', undefined>
type LoadingSuccess = IAction<'LAYOUT_LOADING_SUCCESS', undefined>
type LoadingFail = IAction<'LAYOUT_LOADING_FAIL', { error: any }>

export type Action = InitAction | LoadingBegin | LoadingSuccess | LoadingFail

export const loadingBegin = (): LoadingBegin => ({
  type: 'LAYOUT_LOADING_BEGIN',
  payload: undefined
})

export const loadingSuccess = (): LoadingSuccess => ({
  type: 'LAYOUT_LOADING_SUCCESS',
  payload: undefined
})

export const loadingFail = (error: any): LoadingFail => ({
  type: 'LAYOUT_LOADING_FAIL',
  payload: { error }
})
