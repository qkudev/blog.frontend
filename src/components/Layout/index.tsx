import * as React from 'react'
import { Header, Spinner } from '../index'
import './style'

interface Props {
  loading: boolean
  error?: any
  children?: React.ReactNode
}

function Layout({ children, loading, error }: Props) {
  if (error) {
    return (
      <div className='app'>
        <Header />
        <div id='content' className='layout content'>
          Something went wrong :(
        </div>
      </div>
    )
  }
  if (loading) {
    return <Spinner />
  }
  return (
    <div className='app'>
      <Header />
      <div id='content' className='layout content'>
        {children}
      </div>
    </div>
  )
}

Layout.defaultProps = {
  loading: false
}

export default Layout
