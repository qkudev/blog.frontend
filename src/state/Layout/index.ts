import * as Actions from './actions'

export * from './reducer'
export { Actions }
