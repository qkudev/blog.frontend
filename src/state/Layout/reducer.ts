import { Action } from './actions'

export interface IState extends LoadableState {}

export const initState: IState = {
  loading: false,
  error: undefined
}

const initAction: InitAction = {
  type: '@@INIT',
  payload: undefined
}

export function reducer(
  state: IState = initState,
  action: Action = initAction
): IState {
  switch (action.type) {
    case 'LAYOUT_LOADING_BEGIN':
      return { ...state, loading: true }
    case 'LAYOUT_LOADING_SUCCESS':
      return { ...state, loading: false, error: undefined }
    case 'LAYOUT_LOADING_FAIL':
      return { ...state, loading: false, error: action.payload.error }
    default:
      return state
  }
}
