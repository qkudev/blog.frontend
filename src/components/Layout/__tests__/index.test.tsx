import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'
import Layout from '../index'

describe('Layout', function() {
  test('Matches with snapshot', function() {
    const layout = renderer.create(
      <MemoryRouter>
        <Layout />
      </MemoryRouter>
    )
    let tree = layout.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Matches with error snapshot', function() {
    const errorLayout = renderer.create(
      <MemoryRouter>
        <Layout error='LayoutError' />
      </MemoryRouter>
    )
    let tree = errorLayout.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Matches with loading snapshot', function() {
    const loadingLayout = renderer.create(
      <MemoryRouter>
        <Layout loading={true} />
      </MemoryRouter>
    )
    let tree = loadingLayout.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
