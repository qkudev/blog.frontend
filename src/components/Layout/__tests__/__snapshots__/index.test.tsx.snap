// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Matches with error snapshot 1`] = `
<div
  className="app"
>
  <div
    className="layout header"
    id="app-header"
  >
    <a
      href="/"
      id="header-link"
      onClick={[Function]}
    >
      Show me them all
    </a>
  </div>
  <div
    className="layout content"
    id="content"
  >
    Something went wrong :(
  </div>
</div>
`;

exports[`Layout Matches with loading snapshot 1`] = `
<div
  className="lds-facebook"
>
  <div />
  <div />
  <div />
</div>
`;

exports[`Layout Matches with snapshot 1`] = `
<div
  className="app"
>
  <div
    className="layout header"
    id="app-header"
  >
    <a
      href="/"
      id="header-link"
      onClick={[Function]}
    >
      Show me them all
    </a>
  </div>
  <div
    className="layout content"
    id="content"
  />
</div>
`;
